lib/core/presentation/widgets/movie_card.dart

```diff
@@ -12,18 +18,20 @@ class MovieCard extends StatelessWidget {
       onTap: () {
         Navigator.of(context).pushNamed(Routes.detailsRoute);
       },
-      child: Container(
-        height: 180,
-        width: 135,
-        decoration: BoxDecoration(
-          borderRadius: BorderRadius.circular(15),
-          image: const DecorationImage(
-            fit: BoxFit.cover,
-            image: AssetImage(
-              'asset/poster.jpg',
+      child: CachedNetworkImage(
```

Comment:
You might want to add an error builder for when this image cannot be loaded.
You can simulate it by turning off your internet immediately after loading the movies.